src/main/ui/cheat-code-detector.gd

```diff
@@ -0,0 +1,36 @@
+class_name CheatCodeDetector
+extends Node
+
+signal cheat_detected
+
+const MAX_LENGTH = 32
+const CODE_KEYS := {
+	"a":"", "b":"", "c":"", "d":"", "e":"", "f":"", "g":"", "h":"", "i":"",
+	"j":"", "k":"", "l":"", "m":"", "n":"", "o":"", "p":"", "q":"", "r":"",
+	"s":"", "t":"", "u":"", "v":"", "w":"", "x":"", "y":"", "z":""
+}
+
+export (Array, String) var codes := []
+
+var _pressed_keys: Dictionary = {}
+var _string_buffer: String = ""
```

Comment:
Instead of `string_buffer`, maybe something like `typed_string` or something like that.

Reply:
Changed to `_previous_keypresses`